nimare/base.py

```diff
@@ -283,14 +286,46 @@ def __init__(self, *args, **kwargs):
             mask = get_masker(mask)
         self.masker = mask
 
+        self.resample = kwargs.get("resample", False)
+
+        # defaults for resampling images (nilearn's defaults do not work well)
+        self.resample_kwargs = {
+            k.split("resample__")[1]: v for k, v in kwargs.items() if k.startswith("resample__")
+        } or {"clip": True, "interpolation": "linear"}
+
     def _preprocess_input(self, dataset):
         """Preprocess inputs to the Estimator from the Dataset as needed."""
         masker = self.masker or dataset.masker
+
+        mask_img = masker.mask_img or masker.labels_img
+        if isinstance(mask_img, str):
+            mask_img = nb.load(mask_img)
+
         for name, (type_, _) in self._required_inputs.items():
             if type_ == "image":
+                # If no resampling is requested, check if resampling is required
+                if not self.resample:
+                    check_imgs = {img: nb.load(img) for img in self.inputs_[name]}
+                    check_imgs["reference_masker"] = mask_img
+                    _check_same_fov(**check_imgs, raise_error=True)
```

Comment:
It's pretty trivial, and would only slightly improve readability, but to save the `del` call later you could just pass `reference_masker` as an argument directly to `_check_same_fov` instead of adding it to `check_imgs`, since you remove it right away

Reply:
clean code is happy code :), changed.